diff --git a/src/dactyl_keyboard/dactyl.clj b/src/dactyl_keyboard/dactyl.clj
index 93a2d0c..d73b979 100644
--- a/src/dactyl_keyboard/dactyl.clj
+++ b/src/dactyl_keyboard/dactyl.clj
@@ -13,13 +13,13 @@
 ;; Shape parameters ;;
 ;;;;;;;;;;;;;;;;;;;;;;
 
-(def nrows 4)
-(def ncols 5)
+(def nrows 6)
+(def ncols 6)
 
 (def α (/ π 12))                        ; curvature of the columns
 (def β (/ π 36))                        ; curvature of the rows
 (def centerrow (- nrows 3))             ; controls front-back tilt
-(def centercol 3)                       ; controls left-right tilt / tenting (higher number is more tenting)
+(def centercol 2)                       ; controls left-right tilt / tenting (higher number is more tenting)
 (def tenting-angle (/ π 4))            ; or, change this for more precise tenting control
 (def column-style 
   (if (> nrows 5) :orthographic :standard))  ; options include :standard, :orthographic, and :fixed
@@ -32,7 +32,7 @@
 
 (def thumb-offsets [0 -3 5])
 
-(def keyboard-z-offset 34)              ; controls overall height; original=9 with centercol=3; use 16 for centercol=2
+(def keyboard-z-offset 58)              ; controls overall height; original=9 with centercol=3; use 16 for centercol=2
 
 (def extra-width 2.5)                   ; extra space between the base of keys; original= 2
 (def extra-height 1.0)                  ; original= 0.5
@@ -666,8 +666,8 @@
 (def teensy-holder-height (+ 6 teensy-width))
 (def teensy-offset-height 5)
 (def teensy-holder-top-length 18)
-(def teensy-top-xy (key-position 0       lastrow (map + (wall-locate1  -80.5 -9.5) [(/ mount-width 2) 0 0])))
-(def teensy-bot-xy (key-position lastcol lastrow (map + (wall-locate2  -14.5 -11.6) [(/ mount-width 2) 0 0])))
+(def teensy-top-xy (key-position 0       lastrow (map + (wall-locate1  -84.5 -9.5) [(/ mount-width 2) 0 0])))
+(def teensy-bot-xy (key-position lastcol lastrow (map + (wall-locate2  -11.5 -11.6) [(/ mount-width 2) 0 0])))
 ;(def teensy-holder-length (- (second teensy-top-xy) (second teensy-bot-xy)))
 (def teensy-holder-length teensy-length)
 (def teensy-holder-offset (/ teensy-holder-length -2))
@@ -691,7 +691,7 @@
         ;(translate [(- teensy-holder-width) 0 0])
         ;(translate [7.5 0 0])   ; teensy-holder 7.5mm from the back-wall 
         (rotate (deg2rad  52) [0 0 1])
-        (translate [(+ (first teensy-top-xy) (- teensy-width 6))
+        (translate [(+ (first teensy-top-xy) (+ teensy-width 13))
                     (- (second teensy-top-xy) 13)
                     (/ (+ 6 teensy-width) 2)])
            ))
@@ -722,14 +722,14 @@
                           (if is-right-down
                              (key-position column row (map + (wall-locate3 -2.2 -1.8) [0 (/ mount-height 2) 0]))
                              (if is-right-up
-                                (key-position column row (map - (wall-locate3  -1.6  2.3) [0 (/ mount-height 2) 0]))
+                                (key-position column row (map - (wall-locate3  -1  1.3) [0 (/ mount-height 2) 0]))
                                 (if is-middle-down
-                                   (key-position column row (map + (wall-locate2 -6.3 -5.7) [0 (/ mount-height 2) 0]))
+                                   (key-position column row (map + (wall-locate2 0 -5.7) [0 (/ mount-height 2) 0]))
                                    (if is-middle-up
-                                      (key-position column row (map - (wall-locate2  0  4.5) [0 (/ mount-height 2) 0]))
+                                      (key-position column row (map - (wall-locate2  4.2  3.7) [0 (/ mount-height 2) 0]))
                                       (if is-thumb-up
-                                          (key-position column row (map + (wall-locate1  -80.5 -9.5) [(/ mount-width 2) 0 0]))
-                                          (key-position column row (map + (wall-locate2  -14.5 -11.6) [(/ mount-width 2) 0 0]))
+                                          (key-position column row (map + (wall-locate1  -84.5 -9.5) [(/ mount-width 2) 0 0]))
+                                          (key-position column row (map + (wall-locate2  -11.5 -11.6) [(/ mount-width 2) 0 0]))
                                             ; is-thumb-down
                                       )
                                    )
